src/ucp/core/ucp_ep.c

```diff
@@ -1016,13 +1016,15 @@ void ucp_ep_destroy(ucp_ep_h ep)
 
 int ucp_ep_config_lane_is_equal(const ucp_ep_config_key_t *key1,
                                 const ucp_ep_config_key_t *key2,
-                                ucp_lane_index_t lane, int compare_types)
+                                ucp_lane_index_t lane1,
+                                ucp_lane_index_t lane2,
+                                int compare_types)
 {
-    return (key1->lanes[lane].rsc_index    == key2->lanes[lane].rsc_index)    &&
-           (key1->lanes[lane].proxy_lane   == key2->lanes[lane].proxy_lane)   &&
-           (key1->lanes[lane].dst_md_index == key2->lanes[lane].dst_md_index) &&
-           (key1->lanes[lane].path_index   == key2->lanes[lane].path_index)   &&
-           ((key1->lanes[lane].lane_types  == key2->lanes[lane].lane_types) ||
+    return (key1->lanes[lane1].rsc_index    == key2->lanes[lane2].rsc_index)    &&
+           (key1->lanes[lane1].proxy_lane   == key2->lanes[lane2].proxy_lane)   &&
+           (key1->lanes[lane1].dst_md_index == key2->lanes[lane2].dst_md_index) &&
+           (key1->lanes[lane1].path_index   == key2->lanes[lane2].path_index)   &&
+           ((key1->lanes[lane1].lane_types  == key2->lanes[lane2].lane_types) ||
             !compare_types);
```

Comment:
why needed if it doesn't compare anything?

Reply:
we have two cases where we want to compare `lane_types` and where - we don't:
- https://github.com/openucx/ucx/blob/72190be12dc5c798eb415862774ea15dcd53e8b2/src/ucp/core/ucp_ep.c#L1057
- https://github.com/openucx/ucx/blob/72190be12dc5c798eb415862774ea15dcd53e8b2/src/ucp/wireup/wireup.c#L304